internal/weldr/api.go

```diff
@@ -1883,6 +1915,31 @@ func (api *API) composeHandler(writer http.ResponseWriter, request *http.Request
 		return
 	}
 
+	// Fetch parent ostree commit from ref + url if commit is not
+	// provided. The parameter name "parent" is perhaps slightly misleading
+	// as it represent whatever commit sha the image type requires, not
+	// strictly speaking just the parent commit.
+	if cr.OSTree.Ref != "" && cr.OSTree.URL != "" {
+		if cr.OSTree.Parent != "" {
+			errors := responseError{
+				ID:  "OSTreeOptionsError",
+				Msg: "Supply at most one of Parent and URL",
+			}
+			statusResponseError(writer, http.StatusInternalServerError, errors)
+			return
+		}
+		parent, err := ostreeResolveRef(cr.OSTree.URL, cr.OSTree.Ref)
+		if err != nil {
+			errors := responseError{
+				ID:  "OSTreeCommitError",
+				Msg: err.Error(),
+			}
+			statusResponseError(writer, http.StatusInternalServerError, errors)
```

Comment:
I think that this is also a candidate for `http.BadRequestError`

Reply:
Ditto.